Build textfield components

lukenguyen-fp opened the pull request:
- FormTextField
- DropdownTextField
- PhoneTextField
- SearchTextField

lukenguyen-fp committed:
- Use tableview to optimize dropdown and do calculate estimate height
- Improve phone textfield

lukenguyen-fp committed:
Implement search text field

lukenguyen-fp committed:
Merge branch 'feature/henry-foundation' into feature/luke

henrychu-outcubator commented:
- Need to generic DropdownSheetViewController and DropdownOptionCell to adapt new OptionCellTypes
- Utilize TextFieldFormatter.unformatted to get validationText
- Need to abstract PhoneTextField (consists of 2 parts: prefixDropDownTF & phoneNumberTF)
- SearchTextField should inherit FormTextField instead of being built from scratch

lukenguyen-fp commented on GamePaySDKCoreUI/GamePaySDKCoreUI/Core/Dropdown/DropdownSheetViewController.swift:
```diff
@@ -8,38 +8,37 @@
 import UIKit
 
 class DropdownSheetViewController: UIViewController {
```
Write generic DropdownSheetViewController adopt from BaseCell

lukenguyen-fp commented on GamePaySDKCoreUI/GamePaySDKCoreUI/Core/TextField/Dropdown/DropdownOptionCell.swift:
```diff
@@ -0,0 +1,134 @@
+
+import UIKit
+
+class DropdownOptionCell: UITableViewCell {
```
Write BaseOptionCell

lukenguyen-fp commented on GamePaySDKCoreUI/GamePaySDKCoreUI/Core/TextField/Search/SearchTextField.swift:
```diff
@@ -0,0 +1,168 @@
+
+import UIKit
+
+public class SearchTextField: UIView {
```
Inherit FormTextField instead of new view

lukenguyen-fp committed:
- Implement generic for dropdown
- Utilize text formatter and validation
- Write BaseTextField
- Rewrite SearchTextField

lukenguyen-fp committed:
Do something with base text field

lukenguyen-fp committed:
Refactor dropdown | phone textfield
